airbyte-commons-worker/src/main/java/io/airbyte/workers/internal/DefaultAirbyteStreamFactory.java

```diff
@@ -62,6 +86,23 @@ public Stream<AirbyteMessage> create(final BufferedReader bufferedReader) {
     return bufferedReader
         .lines()
         .peek(str -> metricClient.distribution(OssMetricsRegistry.JSON_STRING_LENGTH, str.length()))
+        .peek(str -> {
+          if (exceptionClass.isPresent()) {
+            long messageSize = str.getBytes(StandardCharsets.UTF_8).length;
+            if (messageSize > maxMemory * 0.8) {
```

Comment:
Would be nice to have some comments about this magic `0.8`

Reply:
I agree, I was questioning that as well. The original ticket was mentioning a 60% ratio. I put 80% out of safety, I wonder if it could make a connection to fail even if it's able to process.

Reply:
I don't think that there is one yet, but we modify a source's `resourceRequirements` and give it a ton of ram.  Parsing the message would work on the source, but not in the worker in that case.  Not that it's likely, but we know that we've encountered users who store binary data (images) in their database, and the source turns them into base64 strings.  That's the kind of message-parsing crash we are trying to avoid with this change.

Reply:
Ok, that PR should prevent that but should we push for the connectors implement that as limitation as well. In this PR a `source error` means that the platform think it won't be able to parse a message from the source. If we want to plug that error, it needs to be directly in the source read method that is call in the IntegrationRunner [here](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/bases/base-java/src/main/java/io/airbyte/integrations/base/IntegrationRunner.java#L138)

Reply:
Agree - but there will always be differences in implementation or OSS sources which don't have this limit in place.

Reply:
Should we consider having a hard limit on the message size?